lib/src/storage/backend/rocksdb.rs

```diff
@@ -370,19 +522,39 @@ impl Db {
     #[must_use]
     pub fn snapshot(&self) -> Reader {
         unsafe {
-            let snapshot = rocksdb_transactiondb_create_snapshot(self.0.db);
-            assert!(
-                !snapshot.is_null(),
-                "rocksdb_transactiondb_create_snapshot returned null"
-            );
             let options = rocksdb_readoptions_create_copy(self.0.read_options);
-            rocksdb_readoptions_set_snapshot(options, snapshot);
-            Reader {
-                inner: InnerReader::Snapshot(Rc::new(InnerSnapshot {
-                    db: self.0.clone(),
-                    snapshot,
-                })),
-                options,
+            match &self.0.db {
+                InnerDbHandler::InMemory(inner_db_handler)
+                | InnerDbHandler::Primary(inner_db_handler) => {
+                    let snapshot = rocksdb_transactiondb_create_snapshot(inner_db_handler.db);
+                    assert!(
+                        !snapshot.is_null(),
+                        "rocksdb_transactiondb_create_snapshot returned null"
+                    );
+                    rocksdb_readoptions_set_snapshot(options, snapshot);
+                    Reader {
+                        inner: InnerReader::Snapshot(Rc::new(InnerSnapshot {
+                            db: self.0.clone(),
+                            snapshot,
+                        })),
+                        options,
+                    }
+                }
+                InnerDbHandler::ReadOnly(_) => Reader {
+                    inner: InnerReader::Secondary(self.0.clone()),
+                    options,
+                },
+                InnerDbHandler::Secondary(_) => {
+                    ffi_result!(rocksdb_try_catch_up_with_primary_with_status(
+                        secondary_db_handler_or_error(&self.0.db).unwrap().db,
+                    ))
+                    .unwrap();
```

Comment:
It might be nicer to return the error here instead of a panic.
Two options:
- change the method signature return a `Result` (may be very invasive in the upper API levels)
- add a `Option<StorageError>` to the returned object storing the error and then, the reader methods return the error (using e.g. the `Option::take` function) if it is set.

An other option might be to just ignore the error completely (or print it to a log) and read from the outdated version. Not sure what would be the best, it depends if we prefer availability or consistency...).

Reply:
i like the second options, but that would add quite some noise to the Reader-implementation? each function would have to be prepended with an error checking function, right?
maybe we can do this in an additional pr?

Reply:
Indeed, it would be quite cumbersome. After thinking about it again, a maybe not too bad approach is maybe to just ignore the errors here by just removing the `.unwrap()`. If things are very bad the following  operation will error anyway. There is already the "dirty read" problem so reading also old data is maybe ok too.